src/mongo/tree/MongoDatabaseTreeItem.ts

```diff
@@ -205,13 +207,14 @@ export class MongoDatabaseTreeItem extends AzureParentTreeItem<IMongoTreeRoot> {
 		}
 	}
 
-	private async createShell(shellPath: string): Promise<Shell> {
-		return <Promise<null>>Shell.create(shellPath, this.connectionString, this.root.isEmulator)
-			.then(
-				shell => {
-					return shell.useDatabase(this.databaseName).then(() => shell);
-				},
-				error => vscode.window.showErrorMessage(error));
+	private async createShell(shellPath: string, shellArgs: string | undefined): Promise<Shell> {
+		let shell = await Shell.create(shellPath, shellArgs, this.connectionString, this.root.isEmulator);
+		let useDatabaseResult = await shell.useDatabase(this.databaseName);
+		if (!useDatabaseResult.match(/^switched to db /i)) {
+			// This might be unexpected output (but maybe it's not in English), show it to the user just in case
+			ext.outputChannel.appendLine(useDatabaseResult);
```

Comment:
As mentioned elsewhere, I think it makes sense to have an outputChannel dedicated to _all_ output from executing mongo scripts